src/Streamly/Internal/Unicode/Stream.hs

```diff
@@ -237,18 +237,21 @@ decodeTable = [
 utf8dLength :: Int
 utf8dLength = length decodeTable
 
+-- | We do not want to garbage collect this and free the memory, we want to
+-- keep this persistent. We don't know how to do that with GHC without having a
+-- reference in some global structure. So we use a hack, use mallocBytes so
+-- that the GC has no way to free it.
 {-# NOINLINE utf8d #-}
 utf8d :: Ptr Word8
 utf8d = unsafePerformIO $ do
-    -- Aligning to cacheline makes a barely noticeable difference
-    -- XXX currently alignment is not implemented for unmanaged allocation
     let size = utf8dLength
     p <- liftIO $ mallocBytes size
     void $ D.fold
         (Fold.foldlM' (\b a -> poke b a >> return (b `plusPtr` 1)) (return p))
         (D.fromList decodeTable)
     return p
 
+-- XXX We can just use Foreign.Storable in this.
```

Comment:
We could, but there was a specific reason for changing this, which I seem to have forgotten. Either way, better to use one class for unboxed representation.